### PR TITLE
fix: check type before casting to PsiDocCommentOwner during hover

### LINKED ISSUE #1102: ClassCastException logged when hovering over a record object in a Qute template
Hovering over `col.charge` in https://github.com/ia3andy/quarkus-blast/blob/main/src/main/resources/templates/GameController/partials/grid.html#L25 results in this exception being logged:
```
2023-08-10 18:23:09,560 [  59721]   WARN - c.r.d.i.q.p.QuteSupportForTemplate - Error while collecting Javadoc for game.Cell#charge
java.lang.ClassCastException: class com.intellij.psi.impl.source.PsiRecordComponentImpl cannot be cast to class com.intellij.psi.PsiDocCommentOwner (com.intellij.psi.impl.source.PsiRecordComponentImpl and com.intellij.psi.PsiDocCommentOwner are in unnamed module of loader com.intellij.ide.plugins.cl.PluginClassLoader @2e58f144)
	at com.redhat.devtools.intellij.quarkus.javadoc.JavadocContentAccess.getHTMLContentReader(JavadocContentAccess.java:23)
	at com.redhat.devtools.intellij.quarkus.javadoc.JavadocContentAccess.getPlainTextContentReader(JavadocContentAccess.java:67)
	at com.redhat.devtools.intellij.lsp4mp4ij.psi.internal.core.ls.PsiUtilsLSImpl.getJavadoc(PsiUtilsLSImpl.java:124)
	at com.redhat.devtools.intellij.qute.psi.QuteSupportForTemplate.getJavadoc(QuteSupportForTemplate.java:415)
	at com.redhat.devtools.intellij.qute.psi.QuteSupportForTemplate.getJavadoc(QuteSupportForTemplate.java:373)
	at com.redhat.devtools.intellij.qute.lsp.QuteLanguageClient.lambda$getJavadoc$13(QuteLanguageClient.java:183)
	at com.redhat.devtools.intellij.lsp4ij.client.LSPCompletableFuture.lambda$nonBlockingReadActionPromise$2(LSPCompletableFuture.java:125)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.insideReadAction(NonBlockingReadActionImpl.java:536)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.lambda$attemptComputation$3(NonBlockingReadActionImpl.java:501)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1154)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.lambda$runInReadActionWithWriteActionPriority$0(ProgressIndicatorUtils.java:75)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runActionAndCancelBeforeWrite(ProgressIndicatorUtils.java:158)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.lambda$runWithWriteActionPriority$1(ProgressIndicatorUtils.java:115)
	at com.intellij.openapi.progress.ProgressManager.lambda$runProcess$0(ProgressManager.java:66)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:188)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:608)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:683)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:639)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:607)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:175)
	at com.intellij.openapi.progress.ProgressManager.runProcess(ProgressManager.java:66)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runWithWriteActionPriority(ProgressIndicatorUtils.java:112)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runInReadActionWithWriteActionPriority(ProgressIndicatorUtils.java:75)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.attemptComputation(NonBlockingReadActionImpl.java:501)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.lambda$transferToBgThread$1(NonBlockingReadActionImpl.java:408)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:833)

```

as confirmed by IJ, it can be a PsiRecordComponentImpl:
Screenshot 2023-08-10 at 18 19 50

### PR DESCRIPTION
**Pull request by fbricon**
Fixes #1102

### CODE REVIEW

File: src/main/java/com/redhat/devtools/intellij/quarkus/javadoc/JavadocContentAccess.java [hunk 1]
```diff
@@ -20,11 +20,16 @@
...
 public class JavadocContentAccess {
     private static Reader getHTMLContentReader(PsiMember member, boolean allowInherited, boolean useAttachedJavadoc) {
-        PsiDocComment doc = ((PsiDocCommentOwner) member).getDocComment();
+        PsiDocComment doc = null;
         PsiElement sourceMember = member.getNavigationElement();
         if (sourceMember instanceof PsiDocCommentOwner) {
             doc = ((PsiDocCommentOwner) sourceMember).getDocComment();
         }
+        if (doc == null) {
+            if (member instanceof PsiDocCommentOwner) {
```
Comment (angelozerr) [inline src/main/java/com/redhat/devtools/intellij/quarkus/javadoc/JavadocContentAccess.java:29]: I will do this test at first and set the code PsiElement sourceMember = member.getNavigationElement();.. in an else

Comment (fbricon) [reply src/main/java/com/redhat/devtools/intellij/quarkus/javadoc/JavadocContentAccess.java:29]: I just kept the same logic as before

Comment (fbricon) [reply src/main/java/com/redhat/devtools/intellij/quarkus/javadoc/JavadocContentAccess.java]: done

**Comment by angelozerr** [comment]
Thanks @fbricon !

**Pull request merged**